ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/PioneerDecryptor.java

```diff
@@ -0,0 +1,52 @@
+package com.mozilla.telemetry.decoder;
+
+import com.mozilla.telemetry.transforms.FailureMessage;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.FlatMapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ProcessFunction;
+import org.apache.beam.sdk.transforms.WithFailures;
+import org.apache.beam.sdk.transforms.WithFailures.Result;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+public class PioneerDecryptor extends
+    PTransform<PCollection<PubsubMessage>, Result<PCollection<PubsubMessage>, PubsubMessage>> {
+
+  public static PioneerDecryptor of() {
```

Comment:
Note on naming that either transforms or their factory methods should be verbs, so this could be `DecryptPioneerPayloads.of` or perhaps `Pioneer.decryptPayloads()` if you ended up wanting to package more things under a single class.